---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)

library(phyloseq)
library(dplyr)
library(tidyr)
library(ggplot2)
```

::: {align="right"}
<img src="https://github.com/AU-ENVS-Bioinformatics/phyloENVS/blob/man/figures/images/au.png" width="2000"/>
:::

# phyloENVS

<!-- badges: start -->

[![DOI](https://zenodo.org/badge/923477160.svg)](https://doi.org/10.5281/zenodo.17053160)

<!-- badges: end -->

The goal of phyloENVS is to provide tailored tools for visualizing and analyzing microbial community data, with a focus on rRNA and mRNA datasets from environmental samples such as soil, ice, and other ecosystems. Built to enhance the functionality of the phyloseq package, it offers customizable and publication-quality visualizations, including ordination, abundance and richness plots for community composition analysis. The package is designed to streamline workflows for researchers within the Environmental Microbiology section at Department of Environmental Science (ENVS) at Aarhus University.

## Installation

You can install the phyloENVS from [GitHub](https://github.com/) with the devtools package.

``` r
# install.packages("devtools")
devtools::install_github("AU-ENVS-Bioinformatics/phyloENVS", build_vignettes = TRUE)
```

## Data Structure

All functions within phyloENVS builds upon the [phyloseq package](https://joey711.github.io/phyloseq/) for microbial community analysis. The phyloseq data structure in R organizes multi-table datasets into a single coherent object. The components of a phyloseq object are listed below and visualized below.

-   OTU Table (otu_table): Abundance data of taxa (read counts) across samples.

-   Taxonomy Table (tax_table): Taxonomic classification (Kingdom to Species) of each contig.

-   Sample Data (sample_data): Metadata for each sample (e.g., location, timepoint, concentration, etc.).

::: {align="center"}
<img src="https://github.com/johanneBiO/phyloENVS/blob/main/man/figures/phyloseq_data_structure.png" width="450"/>
:::

## Getting Started

The vignettes contains a lot of information on how to use phyloENVS. After installation, the vignette can be accessed from the R console by typing:

```{r, eval=FALSE}
browseVignettes("phyloENVS")
```

It is also possible to find an RMarkdown template in RStudio: File -\> New file -\> R Markdown -\> From Template -\> phyloENVS worksheet.

::: {align="center"}
<img src="https://github.com/AU-ENVS-Bioinformatics/phyloENVS/blob/main/man/figures/template.png" width="500"/>
:::

## Example

This is a basic example which shows how you can visualize the relative abundance for a microbial community. The data used comes from ice sampled at the Qaanaaq glacier and at the Greenland ice sheet. We subset the data and only look at transect samples from the Qaanaaq glacier.

```{r}
library(phyloENVS)

data("qaanaaq_rRNA")

qaanaaq_rRNA_sub <- subset_samples(qaanaaq_rRNA,
                                   Transect != "Non-transect") 

abun_plot <- vis_abundance(physeq = qaanaaq_rRNA_sub,
                           group_x = "SampleName",
                           group_split = "Wetness",
                           level_glom = "Phylum",
                           lower_limit = 2)
```

```{r, include = FALSE}
ggsave("man/figures/qaanaaq_abundance.png",
       abun_plot)
```

::: {align="center"}
<img src="https://github.com/AU-ENVS-Bioinformatics/phyloENVS/blob/main/man/figures/qaanaaq_abundance.png" width="750"/>
:::

NMDS (Non-metric Multidimensional Scaling) is a popular technique used for visualizing and interpreting the relationships between samples in high-dimensional datasets. We can look into how different samples cluster based on the available metadata, e.g., wetness.

```{r}
nmds_plot <- vis_nmds(qaanaaq_rRNA_sub,
                      group_color = "Wetness",
                      group_shape = "Transect",
                      encircle = TRUE,
                      fill_circle = TRUE,
                      smooth_circle = 0.05,
                      scale_circle = 0.05,
                      scale_plot = 0.3,
                      circle_edge_size = 0)
```

```{r, include = FALSE}
ggsave("man/figures/qaanaaq_nmds.png",
       nmds_plot)
```

::: {align="center"}
<img src="https://github.com/AU-ENVS-Bioinformatics/phyloENVS/blob/main/man/figures/qaanaaq_nmds.png" width="750"/>
:::

## Details

The available color scales can be seen below. The colors are based on the AU design guide.

::: {align="center"}
<img src="https://github.com/AU-ENVS-Bioinformatics/phyloENVS/blob/main/man/figures/color_scales.png" width="2000"/>
:::

## Acknowledgement

This package builds upon the [phyloseq](https://joey711.github.io/phyloseq/) and [vegan](https://github.com/vegandevs/vegan) packages. Furthermore, this work has been highly inspired by the [RauENVS package](https://github.com/AU-ENVS-Bioinformatics/RauENVS) and shared code from Thanassis Zervas.
