---
title: "Working with mRNA"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{03 Working with mRNA}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This vignette demonstrates how phyloENVS generalize to mRNA data. We will be using mRNA read counts from ice samples collected in Qaanaaq, Greenland, similar to the vignette "Introduction to phyloENVS". We focus on glacier samples collected along two transects. The sampling locations along each transect were categorized into three zones based on wetness. Furthermore, we distinguish between samples facing east (E) and west (W).

## Annotation of mRNA reads using Diamond

We will consider read counts for mRNA transcripts annotated by mapping mRNA contigs to the following three databases:

-   CAZy - Database of carbohydrate-active enzymes, which are involved in the breakdown or modification of complex carbohydrates.

-   NCYc - This database holds genes involved in nitrogen cycling (like those responsible for nitrogen fixation, nitrification, or denitrification).

-   PCyc - Database for phosphorus cycling genes.

```{r setup}
library(phyloENVS)
library(phyloseq)
library(ggplot2)
library(patchwork)

data("qaanaaq_mRNA")
```

For now, we consider differences between samples with respect to the CAZy and PCyc databases due to the limited size of NCyc. We create seperate phyloseq objects for these two selected databases.

```{r}
qaanaaq_mRNA_cazy <- subset_samples(qaanaaq_mRNA,
                                   Transect != "Non-transect") |>
  subset_taxa(Database == "CAZy")

qaanaaq_mRNA_pcyc <- subset_samples(qaanaaq_mRNA,
                                   Transect != "Non-transect") |>
  subset_taxa(Database == "PCyc")
```

### Ordination

We perform Non-metric Multidimensional Scaling (NMDS). We use the [patchwork package](https://patchwork.data-imaginist.com/) to combine multiple plots.

```{r, message=FALSE, fig.width=12, fig.height=5, out.width="100%", fig.align="center"}
vis_nmds(qaanaaq_mRNA_cazy,
         convert_to_rel = TRUE,
         group_color = "Wetness",
         group_shape = "Direction",
         scale_circle = 0.1,
         scale_plot = 0.3) +
  labs(title = "CAZy") +
  vis_nmds(qaanaaq_mRNA_pcyc,
         convert_to_rel = TRUE,
         group_color = "Wetness",
         group_shape = "Direction",
         scale_circle = 0.1,
         scale_plot = 0.3) + 
  labs(title = "PCyc") + 
  plot_layout(guides = "collect")
```

We look at the corresponding Shepard diagrams to assess the goodness of fit for the NMDS plots.

```{r, message=FALSE, fig.width=12, fig.height=5, out.width="100%", fig.align="center"}
vis_shepard(qaanaaq_mRNA_cazy, 
            convert_to_rel = TRUE) +
  labs(title = "CAZy") + 
  vis_shepard(qaanaaq_mRNA_pcyc, 
            convert_to_rel = TRUE) +
  labs(title = "PCyc")
```

### Abundance

In the ordination, the wet samples seemed to be different from the rest. Especially for the PCyc data, the direction might also explain the differences in mRNA expression. We look into the relative abundances for CAZy and PCyc reads.

```{r, fig.width=12, fig.height=5, out.width="100%", fig.align="center"}
vis_abundance(physeq = qaanaaq_mRNA_cazy,
              group_x = "SampleName",
              group_split = "Wetness",
              level_glom = "Level1",
              color_source = "AUjulien",
              lower_limit = 1) +
  labs(title = "CAZy",
       fill = "Level 1") +
  vis_abundance(physeq = qaanaaq_mRNA_pcyc,
              group_x = "SampleName",
              group_split = "Wetness",
              level_glom = "Level1",
              color_source = "AUmort",
              lower_limit = 1) +
  labs(title = "PCyc",
       fill = "Level 1")
```

## Annotation of mRNA reads using AnnoTree

The [ENVS pipeline](https://github.com/AU-ENVS-Bioinformatics/TotalRNA-Snakemake) will also annotate mRNA contigs using [AnnoTree](http://annotree.uwaterloo.ca/annotree/), which includes KEGG, Pfam and Tigrfam annotations for over 30,000 bacterial and 1600 archaeal genomes. Let's load this data.

```{r}
data("qaanaaq_annotree")

qaanaaq_annotree <- subset_samples(qaanaaq_annotree,
                                   Transect != "Non-transect")
```
### Abundance

As before, we visualize the abundance based on the first overall level.

```{r}
vis_abundance(physeq = qaanaaq_annotree,
              group_x = "SampleName",
              group_split = "Wetness",
              level_glom = "Brite_Level1",
              color_source = "AU1",
              lower_limit = 1) +
  labs(fill = "Level 1")
```

The first level shows some very general groupings of the contigs. Let's visualize each one using different color scales to match AU1 above.

```{r}
qaanaaq_annotree |>
  subset_taxa(Brite_Level1 == "Genetic information processing") |>
  vis_abundance(group_x = "SampleName",
                group_split = "Wetness",
                level_glom = "Brite_Level2",
                color_source = "AUblue",
                lower_limit = 5) +
  theme(legend.title=element_blank()) + 
  labs(title = "Genetic information processing") 
```
```{r}
qaanaaq_annotree |>
  subset_taxa(Brite_Level1 == "Metabolism") |>
  vis_abundance(group_x = "SampleName",
                group_split = "Wetness",
                level_glom = "Brite_Level2",
                color_source = "AUturquoise",
                lower_limit = 5) +
  theme(legend.title=element_blank()) + 
  labs(title = "Metabolism")
```

```{r}
qaanaaq_annotree |>
    subset_taxa(Brite_Level1 == "Signaling and cellular processes") |>
    vis_abundance(group_x = "SampleName",
                  group_split = "Wetness",
                  level_glom = "Brite_Level2",
                  color_source = "AUorange",
                  lower_limit = 5) +
    theme(legend.title=element_blank()) + 
    labs(title = "Signaling and cellular processes")
```
